Mathlib/Topology/Category/Top/Limits/Pullbacks.lean

```diff
@@ -370,43 +390,147 @@ theorem pullback_fst_image_snd_preimage (f : X ⟶ Z) (g : Y ⟶ Z) (U : Set Y)
   · rintro ⟨y, hy, rfl⟩
     exact
       ⟨(pullback.snd : pullback f g ⟶ _) y, hy,
-        (concrete_category.congr_hom pullback.condition y).symm⟩
+        (ConcreteCategory.congr_hom pullback.condition y).symm⟩
   · rintro ⟨y, hy, eq⟩
-    exact ⟨(TopCat.pullbackIsoProdSubtype f g).inv ⟨⟨_, _⟩, Eq.symm⟩, by simpa, by simp⟩
+    exact ⟨(TopCat.pullbackIsoProdSubtype f g).inv ⟨⟨_, _⟩, eq.symm⟩, by simpa, by simp⟩
 #align Top.pullback_fst_image_snd_preimage TopCat.pullback_fst_image_snd_preimage
 
 end Pullback
 
-theorem coinduced_of_isColimit {F : J ⥤ TopCat.{max v u}} (c : Cocone F) (hc : IsColimit c) :
-    c.pt.TopologicalSpace = ⨆ j, (F.obj j).TopologicalSpace.coinduced (c.ι.app j) := by
-  let homeo := homeo_of_iso (hc.cocone_point_unique_up_to_iso (colimit_cocone_is_colimit F))
+/-- The binary coproduct cofan in `TopCat`. -/
+protected def binaryCofan (X Y : TopCat.{u}) : BinaryCofan X Y :=
+  BinaryCofan.mk (⟨Sum.inl, by continuity⟩ : X ⟶ TopCat.of (Sum X Y)) ⟨Sum.inr, by continuity⟩
+#align Top.binary_cofan TopCat.binaryCofan
+
+/-- The constructed binary coproduct cofan in `TopCat` is the coproduct. -/
```

Comment:
Are the next 100 lines new? Or did this move in ml3 during the porting of this file? I don't see it when comparing to the mathport output that this port started from...

Reply:
Ah. These should be at the end of `Products.lean`. I've moved them back.